### ISSUE TITLE
[Bug]:  [Bug]: Buttons should be aria-disabled instead of disabled

### ISSUE THREAD

**Issue by Sidnioulz** [storybookjs/storybook#31678]
### Describe the bug

An HTML button that is `disabled` is removed from the accessibility object model altogether and cannot be reached via keyboard navigation. This means that sighted users get feedback that a feature exists but is unavailable, whereas non-sighted users do not know the feature exists at all.

Instead, it's recommended to:
* use `aria-disabled` so it's communicated that the button exists but cannot be used
* use CSS to disable the cursor and pointer events
* add a condition in the click handler to not call the passed onClick prop if the button is disabled
* provide feedback as to why a button is disabled

Therefore, I propose to:
* Replace the `disabled` prop in Button's JSX markup with `aria-disabled`, based on the `disabled` prop's value
* Allow a `string` type for disabled, which gets displayed as a tooltip and `aria-describedby` attribute and explains why the button is disabled and what must be done to enable it

### Reproduction link

yarn storybook:ui in core repo

### Reproduction steps

1. Navigate to a UI with disabled buttons
2. Load up your favourite screenreader
3. Turn off your screen and try to make sense of the UI

### System

```bash
Storybook Environment Info:

  System:
    OS: Linux 6.14 Arch Linux
    CPU: (28) x64 Intel(R) Core(TM) i7-14700KF
    Shell: 5.2.37 - /usr/bin/bash
  Binaries:
    Node: 20.19.0 - ~/.config/nvm/versions/node/v20.19.0/bin/node
    Yarn: 4.9.1 - /usr/bin/yarn <----- active
    npm: 10.8.2 - ~/.config/nvm/versions/node/v20.19.0/bin/npm
```

### Additional context

See relevant articles:

- [Deque - Difference between both disabled attributes](https://www.deque.com/blog/distinguishing-between-aria-and-native-html-attributes/)
- [Smashing - Frustrating design patterns: Disabled buttons](https://www.smashingmagazine.com/2021/08/frustrating-design-patterns-disabled-buttons/)
- [CSS Tricks - Making disabled buttons more inclusive](https://css-tricks.com/making-disabled-buttons-more-inclusive/)
- [Axesslab - Disabled buttons suck](https://axesslab.com/disabled-buttons-suck/)
- [MDN - Aria-disabled](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-disabled)

**Comment by WioletaKolodziej** [comment]
Hi @Sidnioulz 
I wanted to try contributing and went ahead and added my PR to this issue before being officially assigned. 
I’d be very grateful for a review of the changes.

I’ve run yarn test locally and didn’t find any failing dependencies or tests that would cause errors - if there are any other ways to check that, I'd be happy to familiarize myself with them.

**Comment by Sidnioulz** [comment]
> Hi [@Sidnioulz](https://github.com/Sidnioulz) I wanted to try contributing and went ahead and added my PR to this issue before being officially assigned. I’d be very grateful for a review of the changes.
> 
> I’ve run yarn test locally and didn’t find any failing dependencies or tests that would cause errors - if there are any other ways to check that, I'd be happy to familiarize myself with them.

Thanks for giving this a go!

Storybook relies a lot on stories for testing, so it's always a good idea to run Storybook locally while making UI changes. Here, the general process is:
* (once) `yarn install`
* (once) `yarn task compile`
* `cd code && yarn build --watch storybook`
* (in other tab, in `code`, rerun to see changes outside of stories) `yarn storybook:ui`

See the PR for PR feedback!
